Source/Managers/DFU/FirmwareUpgradeManager.swift

```diff
@@ -62,8 +62,9 @@ public class FirmwareUpgradeManager : FirmwareUpgradeController, ConnectionObser
             log(msg: "Firmware upgrade is already in progress", atLevel: .warning)
             return
         }
-        imageData = data
-        hash = try McuMgrImage(data: imageData).hash
+        
+        i = 0
+        images = try [FirmwareUpgradeImage(slot: 0, image: data)]
```

Comment:
The old `start(data:)` method could just call the new `start(images:)` method, instead of duplicating code.

Reply:
Fixed.